File: 4332-CheckGoodInteger/4332-CheckGoodInteger.py
# Last updated: 12/08/2026, 11:31:43
class Solution:
    def checkGoodInteger(self, n: int) -> bool:

        digit_sum = 0
        square_sum = 0

        for digit in str(n):
            d = int(digit)

            digit_sum += d
            square_sum += d*d
            
        if(square_sum - digit_sum) >= 50:
            return True
        else:
            return False

File: 4332-CheckGoodInteger/4332-CheckGoodInteger.java
// Last updated: 09/07/2026, 10:35:35
class Solution {
    private int digitSum(int num) {
        int sum = 0;
        while(num != 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }

    private int squareSum(int num) {
        int sum = 0;
        while(num != 0) {
            sum += (num % 10) * (num % 10);
            num /= 10;
        }
        return sum;
    }
    
    public boolean checkGoodInteger(int n) {
        return (squareSum(n) - digitSum(n)) >= 50;
    }
}